app/ui/parse_resume.py

```diff
@@ -51,6 +52,32 @@ def __init__(self,
         self.hard_skills = set(hard_skills)
         self.nlp = spacy.load('en')
 
+    def get_skills_hard(self, input_text):
+        """
+        Given a string with an input document, return an array
+        of "skills".
+
+        Args:
+            input_text (str): A string representing some document
+                              like a resume
+        """
+
+        # check inputs
+        assert isinstance(input_text, str)
+
+        in_text = input_text
+        for r in [' ', ',', ';', ':', '\n', '!']:
+            if r in in_text:
+                in_text = in_text.replace(r, " ")
+
+        input_words = in_text.split(" ")
+        print("input words:{}".format(input_words))
+        input_words_lower = [word.lower().replace(".", "").strip() if "." in word.lower() else word.lower().strip() for word in input_words]
```

Comment:
this is super hard to read. Also it's weird to me that we're doing this `.` replacement instead of just handling it in the block above.

```
in_text = in_text.replace(".", "")
input_words = in_text.split(" ")
input_words = list(map(lambda x: x.lower(), input_words))
```

Reply:
we want to do a `.` replacement only on a single token/word and not on the entire string. If you do it on the entire string (`in_text`), you would end up replacing valid skills like `.net`, etc. But, I just realized what I have is not correct - what I intended to do was - 
```[word.lower().replace(".", "").strip() if word.lower().endswith(".") else word.lower().strip() for word in input_words]
```